paper-api/src/main/java/org/bukkit/event/player/PlayerRespawnEvent.java

```diff
@@ -27,41 +24,16 @@ public PlayerRespawnEvent(@NotNull final Player respawnPlayer, @NotNull final Lo
 
     @Deprecated(since = "1.19.4")
     public PlayerRespawnEvent(@NotNull final Player respawnPlayer, @NotNull final Location respawnLocation, final boolean isBedSpawn, final boolean isAnchorSpawn) {
-        this(respawnPlayer, respawnLocation, isBedSpawn, false, RespawnReason.PLUGIN);
+        this(respawnPlayer, respawnLocation, isBedSpawn, isAnchorSpawn, RespawnReason.PLUGIN);
     }
 
-    @Deprecated // Paper
+    @ApiStatus.Internal
     public PlayerRespawnEvent(@NotNull final Player respawnPlayer, @NotNull final Location respawnLocation, final boolean isBedSpawn, final boolean isAnchorSpawn, @NotNull final RespawnReason respawnReason) {
-        // Paper start
-        this(respawnPlayer, respawnLocation, isBedSpawn, isAnchorSpawn, respawnReason, com.google.common.collect.ImmutableSet.builder());
-    }
-
-    public PlayerRespawnEvent(@NotNull final Player respawnPlayer, @NotNull final Location respawnLocation, final boolean isBedSpawn, final boolean isAnchorSpawn, @NotNull final RespawnReason respawnReason, @NotNull final com.google.common.collect.ImmutableSet.Builder<org.bukkit.event.player.PlayerRespawnEvent.RespawnFlag> respawnFlags) {
-        // Paper end
-        super(respawnPlayer);
-        this.respawnLocation = respawnLocation;
-        this.isBedSpawn = isBedSpawn;
-        this.isAnchorSpawn = isAnchorSpawn;
-        this.respawnReason = respawnReason;
-        // Paper start
-        if (this.isBedSpawn) { respawnFlags.add(RespawnFlag.BED_SPAWN); }
-        if (this.isAnchorSpawn) { respawnFlags.add(RespawnFlag.ANCHOR_SPAWN); }
-        this.respawnFlags = respawnFlags.build();
-        // Paper end
-    }
-
-    /**
-     * Gets the current respawn location
-     *
-     * @return Location current respawn location
-     */
-    @NotNull
-    public Location getRespawnLocation() {
-        return this.respawnLocation;
```

Comment:
PlayerRespawnEvent supports mutability, this change removes that behavior as AbstractRespawnEvent clones it.

Reply:
This was discussed, there is a setter for the respawn location, so this is intential bulletproofing.

Reply:
I guess, just not sure about doing changes like these silently.